internal/build/job/maker.go

```diff
@@ -113,14 +122,48 @@ func (m *maker) MakeJob(mod ootov1alpha1.Module, buildConfig *ootov1alpha1.Build
 	return job, nil
 }
 
-func MakeSecretVolumes(secretRefs []v1.LocalObjectReference) []v1.Volume {
+func (m *maker) makeImagePullSecretVolume(secretRef v1.LocalObjectReference) v1.Volume {
+
+	vol := v1.Volume{
+		Name: volumeNameFromSecretRef(secretRef),
+		VolumeSource: v1.VolumeSource{
+			Secret: &v1.SecretVolumeSource{
+				SecretName: secretRef.Name,
+				Items: []v1.KeyToPath{
+					{
+						Key:  v1.DockerConfigJsonKey,
+						Path: "config.json",
+					},
+				},
+			},
+		},
+	}
+
+	return vol
+}
+
+func (m *maker) makeImagePullSecretVolumeMount(secretRef v1.LocalObjectReference) v1.VolumeMount {
+
+	volMount := v1.VolumeMount{
+		Name:      volumeNameFromSecretRef(secretRef),
+		ReadOnly:  true,
+		MountPath: "/kaniko/.docker",
+	}
+
+	return volMount
+}
+
+func (m *maker) makeBuildSecretVolumes(secretRefs []v1.LocalObjectReference) []v1.Volume {
```

Comment:
I don't think this needs to be a method.

Reply:
Agree. Fixing.